docs/prompts.md

```diff
@@ -42,65 +42,81 @@ generate({
-which can be used with any model. Optionally, you can also define an input schema
-for the prompt, which is analagous to the input schema for a flow.
-Prompts can also define any of the common model configuration options, such as
-temperature or number of output tokens.
+and then call the prompt via
+
+```ts
+helloPrompt.generate({ input: { name: 'Fred' } });
+// Example output: Hello Fred! 👋 It's nice to meet you. How can I help you today? 😊
+```
 
-You can use this prompt in your code with the `renderPrompt()` helper function.
-Provide the input variables expected by the prompt, and the model to call.
+As shown above, prompts defined this way can specify the structured inputs they accept through the `input.schema` configuration.  This allows you a typesafe way to ensure that prompts can only be invoked with valid sets of inputs.
 
-```javascript
-import { generate, render } from '@genkit-ai/ai';
+Dotprompts can also specify an output, which they will pass along to call to the LLM as a directive (either as an in-context message or as an API parameter for LLMs which support a structured output mode). This guarantees that you'll either get a conforming response, or an exception you can deal with cleanly.
 
-generate(
-  renderPrompt({
-    prompt: helloPrompt,
-    input: { name: 'Fred' },
+```ts
+export const threeGreetingsPrompt = defineDotprompt(
+  {
+    name: 'threeGreetingsPrompt',
     model: 'googleai/gemini-1.5-flash-latest',
-  })
+    input: {
+      schema: z.object({ name: z.string() }),
+    },
+    output: {
+      schema: z.object({
+        short: z.string(),
+        friendly: z.string(),
+        likeAPirate: z.string()
+      })    
+    }
+  },
+  `You are a helpful AI assistant named Walt. Say hello to {{name}}, write a response for each of the styles requested`
 );
 ```
 
-In the Genkit Developer UI, you can run any prompt you have defined in this way.
-This allows you to experiment with individual prompts outside of the scope of
+You can then call `generate` on that prompt and work with the structred output in the response
+
+```ts
+threeGreetingsPrompt.generate({
+   input: { name: 'Fred' } }).output();
+/* Example output
+{
+  "short": "Hello Fred.",
+  "friendly": "Hi Fred, how are you doing today?",
+  "likeAPirate": "Ahoy there, Fred!  May the winds be ever in your favor!"
+}
+*/
+```
+
+In the Genkit Developer UI, you can run any prompt you have defined in this way.  This allows you to experiment with individual prompts outside of the scope of
 the flows in which they might be used.
 
+<img src="resources/prompts-in-developer-ui.png" alt="The developer UI showing JSON response to the threeGreetingsPrompt" class="screenshot-attempt-right">
+
 ## Dotprompt
 
-Genkit includes the [Dotprompt](./dotprompt.md) library which adds additional
-functionality to prompts.
+See the [Dotprompt](./dotprompt.md) page for more features of the Dotprompt library, including
```

Comment:
Why not consolidate these pages into one? You've already started to do that by introducing input and output schema conformance above, which is also listed below as an advanced topic to be covered later!

Reply:
My thinking is that `dotprompt` is too complex for the intro doc since it covers many usecases, so it's better to have one intro page for prompting in general and a deep-dive page for dotprompt

Reply:
I would like to try to consolidate the pages to see how it looks but I can take it on myself in a follow-up PR